src/components/Frame/components/ContextualSaveBar/tests/ContextualSaveBar.test.tsx

```diff
@@ -22,50 +23,149 @@ describe('<ContextualSaveBar />', () => {
+        };
 
-      const contextualSaveBar = mountWithAppProvider(
-        <ContextualSaveBar discardAction={discardAction} />,
-      );
+        const contextualSaveBar = mountWithAppProvider(
+          <ContextualSaveBar discardAction={discardAction} />,
+        );
 
-      contextualSaveBar.find(Button).simulate('click');
-      expect(discardAction.onAction).toHaveBeenCalled();
-    });
+        contextualSaveBar.find(Button).simulate('click');
+        expect(discardAction.onAction).toHaveBeenCalled();
+      });
 
-    it('does not call the discardAction when discardConfirmationModal is true', () => {
-      const discardAction = {
-        content: 'Discard',
-        onAction: jest.fn(),
-        discardConfirmationModal: true,
-      };
+      it('does not render a DiscardConfirmationModal', () => {
+        const discardAction = {
+          content: 'Discard',
+          onAction: noop,
+          discardConfirmationModal: false,
+        };
 
-      const contextualSaveBar = mountWithAppProvider(
-        <ContextualSaveBar discardAction={discardAction} />,
-      );
+        const contextualSaveBar = mountWithAppProvider(
+          <ContextualSaveBar discardAction={discardAction} />,
+        );
 
-      contextualSaveBar.find(Button).simulate('click');
-      expect(discardAction.onAction).not.toHaveBeenCalled();
+        expect(contextualSaveBar.find(DiscardConfirmationModal)).toHaveLength(
+          0,
+        );
+      });
     });
 
-    it('opens a modal with the discardAction when discardConfirmationModal is true', () => {
-      const discardAction = {
-        content: 'Discard',
-        onAction: jest.fn(),
-        discardConfirmationModal: true,
-      };
+    describe('discardConfirmationModal is true', () => {
+      it('does not call the discardAction when the discard button is clicked', () => {
+        const discardAction = {
+          content: 'Discard',
+          onAction: jest.fn(),
+          discardConfirmationModal: true,
+        };
 
-      const contextualSaveBar = mountWithAppProvider(
-        <ContextualSaveBar discardAction={discardAction} />,
-      );
+        const contextualSaveBar = mountWithAppProvider(
+          <ContextualSaveBar discardAction={discardAction} />,
+        );
+
+        contextualSaveBar.find(Button).simulate('click');
+        expect(discardAction.onAction).not.toHaveBeenCalled();
+      });
+
+      it('renders a DiscardConfirmationModal with an `open` prop set to false', () => {
+        const discardAction = {
+          content: 'Discard',
+          onAction: noop,
+          discardConfirmationModal: true,
+        };
+
+        const contextualSaveBar = mountWithAppProvider(
+          <ContextualSaveBar discardAction={discardAction} />,
+        );
+
+        const discardConfirmationModal = contextualSaveBar.find(
+          DiscardConfirmationModal,
+        );
+        expect(discardConfirmationModal).toHaveLength(1);
```

Comment:
I don't think we need this expect since `prop(X)` can only be run on a single node